modules/internal/ArrayViewRankChange.chpl

```diff
@@ -470,6 +762,75 @@ module ArrayViewRankChange {
     proc _getRCREView() {
       return this;
     }
+
+    proc dsiDestroyArr(isalias:bool) {
+      if ownsArrInstance {
+        _delete_arr(_ArrInstance, _isPrivatized(_ArrInstance));
+      }
+    }
+  }  // end of class ArrayViewRankChangeArr
+
+
+  // helper routines for converting things from higher- to
+  // lower-dimensions and back again
+
+  inline proc chpl_rankChangeConvertIdx(i: integral, collapsedDim, idx) {
+    param downrank = collapsedDim.size;
+    var ind = idx;
+    var j = 1;
+    for param d in 1..downrank {
+      if !collapsedDim(d) {
+        ind(d) = i;
+        j += 1;
+      }
+    }
+    assert (j == 2);
```

Comment:
There seem to be too many of these methods. Could they call each other to make the relationship clearer? E.g. is this the i-isnt-a-tuple version? When is that called? If i isn't a tuple, is the number of dimensions necessarily 2?

And then, shouldn't the methods around line 260 such as chpl_rankChangeConvertLoDTupleToHiD, chpl_rankChangeConvertHiDTupleToLoD, downIdxToUpIdx just call methods down here?